app/services/uuid_reporter.rb

```diff
@@ -0,0 +1,134 @@
+class UuidReporter
+  def self.run(**opts)
+    new(**opts).run
+  end
+
+  # Instantiate the service object
+  #
+  # @attr email_file [String] the location of the file with the list of emails
+  # @attr sp_file [String] the location of the file with the list of SP issuers
+  # @attr output [String] the location where the output CSV should be written
+  def initialize(email_file:, sp_file:, output:)
+    @emails = parse_file(email_file).map(&:downcase).uniq
+    validate_emails
+    @issuers = parse_file(sp_file).uniq
+    validate_issuers
+    @output = output
+    validate_output
+  end
+
+  # Take the lists of ServiceProvider issuers and emails and do the following:
+  #   1. Confirm that the user has an Identity with at least one of the SPs
+  #   2. Find the user's associated AgencyIdentity and uuid
+  #   3. Export a CSV file with email addresses and associated uuids
+  #
+  # @return [Integer] the number of uuids collected
+  def run
+    @agency = find_agency
+    @user_ids = collect_user_ids
```

Comment:
I would call this `email_to_user_id` or something, it's not clear to me unless I look closely that it's got the emails

Reply:
Makes sense, thanks!